File: 14-longest-common-prefix/longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        int n = strs.length;
        int m = Integer.MAX_VALUE;

        for (int i = 0; i < n; i++) {
            m = Math.min(strs[i].length(), m);
        }

        String res = "";

        for (int i = 0; i < m; i++) {
            char c = strs[0].charAt(i);

            for (int k = 0; k < n; k++) {
                if (strs[k].charAt(i) != c) {
                    return res;
                }
            }
            res += c;
        }

        return res;
    }
}

File: 14-longest-common-prefix/longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        min_len = len(min(strs,key=len))
        res = ""

        for i in range(min_len):
            for j in range(len(strs)):
                if strs[j][i] != strs[0][i]:
                    return res; 
            
            res = res+strs[j][i]

        return res;         
                